set.mm

```diff
@@ -38138,21 +38163,50 @@ technically classes despite morally (and provably) being sets, like ` 1 `
       BCLM $.
   $}
 
+  ${
+    $d x y z ph $.  $d y z ch $.  $d y z ps $.
+    ss2abdv.1 $e |- ( ph -> ( ps -> ch ) ) $.
+    $( Deduction of abstraction subclass from implication.  (Contributed by NM,
+       29-Jul-2011.)  Avoid ~ ax-8 , ~ ax-10 , ~ ax-11 , ~ ax-12 .  (Revised by
+       Gino Giotto, 28-Jun-2024.) $)
+    ss2abdv $p |- ( ph -> { x | ps } C_ { x | ch } ) $=
+      ( vy vz cab cin wceq wss cv wcel wa df-in wsb df-clab bicomi sbequ wi weq
+      anbi12i abbii anbi12d sbievw ax-1 impd sbimdv ancld impbid syl5bb 3bitr4g
+      a1i eqrdv syl5eqr syl5eq df-ss sylibr ) ABDHZCDHZIZUSJUSUTKAVAFLZUSMZVBUT
+      MZNZFHZUSFUSUTOAVFBDFPZCDFPZNZFHZUSVIVEFVGVCVHVDVCVGBFDQRVDVHCFDQRUBUCAGV
+      JUSAVIFGPZBDGPZGLZVJMVMUSMVKVLCDGPZNZAVLVIVOFGFGUAVGVLVHVNBFGDSCFGDSUDUEA
+      VOVLAVLVNVLVLVNVLTTAVLVNUFUMUGAVLVNABCDGEUHUIUJUKVIGFQBGDQULUNUOUPUSUTUQU
+      R $.
+    $( $j usage 'ss2abdv' avoids 'ax-8' 'ax-10' 'ax-11' 'ax-12'; $)
+  $}
+
+  ${
+    $d x ph $.
+    ss2abdvOLD.1 $e |- ( ph -> ( ps -> ch ) ) $.
+    $( Obsolete version of ~ ss2abdv as of 28-Jun-2024.  (Contributed by NM,
+       29-Jul-2011.)  (Proof modification is discouraged.)
+       (New usage is discouraged.) $)
+    ss2abdvOLD $p |- ( ph -> { x | ps } C_ { x | ch } ) $=
+      ( wi wal cab wss alrimiv ss2ab sylibr ) ABCFZDGBDHCDHIAMDEJBCDKL $.
+  $}
```

Comment:
I would like to keep this theorem (maybe as ss2abdvALT) because its proof is so much shorter than the new one.

Reply:
How about keeping @icecream17 proof instead:

<pre>
  ${
    $d x y ph $.  $d y ps $.  $d y ch $.
    ss2abdvALT.1 $e |- ( ph -> ( ps -> ch ) ) $.
    $( ALternate version of ~ ss2abdv .  Shorter, but requiring ~ ax-8 .
       (Contributed by Steven Nguyen, 28-Jun-2024.)
       (Proof modification is discouraged.)  (New usage is discouraged.) $)
    ss2abdvALT $p |- ( ph -> { x | ps } C_ { x | ch } ) $=
      ( vy cab wsb cv wcel sbimdv df-clab 3imtr4g ssrdv ) AFBDGZCDGZABDFHCDFHFI
      ZOJQPJABCDFEKBFDLCFDLMN $.
  $}
</pre>

His version is not that much longer than the original and it does not use ax-10, ax-11, ax-12.

Reply:
Yes, this is a good suggestion. Then this theorem (version) can/should be added.

Reply:
The comment contains a typo 'ALternate...'

Reply:
Done